### ISSUE TITLE
Pester test runner: Default variables not applicable to release builds and needs more input validation

### ISSUE THREAD

**Issue by bergmeister** [rfennell/vNextBuild#350]
### VSTS Extension you are using
Pester test runner v8

### Where are you running it?
VSTS

### Version of Extension/Task
v8

### Expected behaviour and actual behaviour

When adding the task, it pre-populates some of the fields with defaults and some of them contain VSTS variables that are only available in build definitions, such as `$(Build.SourcesDirectory)`, which is not available in release definitions. Either a variables should be chosen that is available in both build and release or possibly a variable such as `$(System.DefaultWorkingDirectory)` should be chosen for release definitions.
![image](https://user-images.githubusercontent.com/9250262/42453260-186e1d4c-8384-11e8-9d02-ce46c6fb819d.png)
When it failed, it is was impossible to diagnose what went wrong, therefore more input validation would be useful

```
2018-07-09T13:15:23.1189871Z ##[section]Starting: Pester Test Runner
2018-07-09T13:15:23.1391553Z ==============================================================================
2018-07-09T13:15:23.1391819Z Task         : Pester Test Runner from Black Marble
2018-07-09T13:15:23.1392123Z Description  : Run Pester tests by either installing the latest version of Pester at run time (if possible) or using the version shipped with the task (4.3.1)
2018-07-09T13:15:23.1392382Z Version      : 8.1.2
2018-07-09T13:15:23.1392732Z Author       : Richard Fennell
2018-07-09T13:15:23.1393247Z Help         : Version: 8.1.2. [More Information](https://github.com/rfennell/vNextBuild/wiki/Pester-Task/)
2018-07-09T13:15:23.1393466Z ==============================================================================
2018-07-09T13:15:23.1708356Z Preparing task execution handler.
2018-07-09T13:15:23.7373209Z Executing the powershell script: D:\_Builds\RCSSV-EM20-005T.01\_tasks\Pester_31ef0033-64e3-4c55-b888-f446541474a6\8.1.2\Pester.ps1
2018-07-09T13:15:23.7490710Z ##[error]System.Management.Automation.ParameterBindingValidationException: Cannot validate argument on parameter 'resultsFile'. The "
2018-07-09T13:15:23.7499038Z             (Test-Path (Split-Path $_ -Parent)) -and ($_.split('.')[-1] -eq 'xml')
2018-07-09T13:15:23.7499538Z         " validation script for the argument with value "$(Build.SourcesDirectory)\PesterPowerShellCmdletsSystemTests.XML" did not return a result of True. Determine why the validation script failed, and then try the command again. ---> System.Management.Automation.ValidationMetadataException: The "
2018-07-09T13:15:23.7499967Z             (Test-Path (Split-Path $_ -Parent)) -and ($_.split('.')[-1] -eq 'xml')
2018-07-09T13:15:23.7500369Z         " validation script for the argument with value "$(Build.SourcesDirectory)\PesterPowerShellCmdletsSystemTests.XML" did not return a result of True. Determine why the validation script failed, and then try the command again.
2018-07-09T13:15:23.7501402Z    at System.Management.Automation.ValidateScriptAttribute.ValidateElement(Object element)
2018-07-09T13:15:23.7501757Z    at System.Management.Automation.ParameterBinderBase.BindParameter(CommandParameterInternal parameter, CompiledCommandParameter parameterMetadata, ParameterBindingFlags flags)
2018-07-09T13:15:23.7502217Z    --- End of inner exception stack trace ---
2018-07-09T13:15:23.7502399Z    at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
2018-07-09T13:15:23.7502644Z    at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
2018-07-09T13:15:23.7502887Z    at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
2018-07-09T13:15:23.7503174Z    at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2018-07-09T13:15:23.7503468Z    at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
2018-07-09T13:15:23.7503760Z    at Microsoft.TeamFoundation.DistributedTask.Handlers.LegacyVSTSPowerShellHost.VSTSPowerShellHost.Main(String[] args)
2018-07-09T13:15:23.7511784Z ##[error]LegacyVSTSPowerShellHost.exe completed with return code: -1.
2018-07-09T13:15:23.7521745Z ##[section]Finishing: Pester Test Runner
```

**Comment by ChrisLGardner** [comment]
Thanks for this, I get it sorted this week. I think the System.WorkingDirectory exists in a build too so I'll swap to that once I check the docs.
________________________________

**Comment by ChrisLGardner** [comment]
Released in 8.2.x
